dts/bindings/gpio/ambiq,gpio.yaml

```diff
@@ -0,0 +1,19 @@
+# Copyright (c) 2023 Antmicro <www.antmicro.com>
+# SPDX-License-Identifier: Apache-2.0
+
+description: Ambiq GPIO
+
+compatible: "ambiq,gpio"
+
+include: [gpio-controller.yaml, base.yaml]
+
+properties:
+  reg:
+    required: true
```

Comment:
Please add a description of how the `reg` property is used for the child devices and its relationship with the parent `reg` property.